### PR TITLE
Improved documentation versioning

### PR DESCRIPTION
**Pull request by TOsmanov**
#### This PR contain
- updated `site/build_docs.py` thus, when using `git checkout` from another version, only the "docs" and "Images" folders are added, and not the entire structure, this is necessary for the correct operation of the rest of the site.
- add `navbar-version-selector.html` to ensure that when you change the version, the current page in a different version opens and not the root of the site
- in `_index.html` adding meta tag for redirecting to about page https://tosmanov.github.io/cvat/

#### Example of site generation
https://tosmanov.github.io/cvat/docs/

#### Current issue 
- [ ] page about is duplicated in each version
- [ ] there is no warning banner of archived versions

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: site/layouts/partials/navbar-version-selector.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{{ $page_rel := .Page.RelPermalink | safeURL }}
+<a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+	{{ .Site.Params.version_menu }}
+</a>
+<div class="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
+	{{ range .Site.Params.versions }}
+		{{ $ver := .version }}
```
Comment (azhavoro) [inline site/layouts/partials/navbar-version-selector.html:7]: Please replace all tabs with spaces.
P.S. VS Code can be easily configured to automate this: https://code.visualstudio.com/docs/editor/codebasics#_indentation

File: site/layouts/partials/navbar-version-selector.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+			{{ if eq $ver "develop" }}
+				{{ $page_rel_trim := strings.TrimLeft "/" $page_rel | safeURL }}
+					<a class="dropdown-item" href="{{ .url }}{{ $page_rel_trim }}">{{ $ver }}</a>
+			{{ else }}
+				<a class="dropdown-item" href="{{ .url }}{{ $page_rel }}">{{ $ver }}</a>
+			{{ end }}
```
Comment (azhavoro) [inline site/layouts/partials/navbar-version-selector.html:13]:
```suggestion
			{{ if eq $ver "develop" }}
				{{ $page_rel := strings.TrimLeft "/" $page_rel | safeURL }}
			{{ end }}
			<a class="dropdown-item" href="{{ .url }}{{ $page_rel }}">{{ $ver }}</a>
```

Will it work?

Comment (TOsmanov) [reply site/layouts/partials/navbar-version-selector.html]: This does not work, I assume that line 11 is outside the loop {{ if eq $ver "develop" }} and the variable is taken from line 1

Comment (azhavoro) [reply site/layouts/partials/navbar-version-selector.html]: `{{ $page_rel = strings.TrimLeft "/" $page_rel | safeURL }}`

Ok, I'm not familiar with go, but looks like `=` operator should be used here https://gohugo.io/templates/introduction/#variables. Could you please try?

Comment (TOsmanov) [reply site/layouts/partials/navbar-version-selector.html]: it looks like in this case the variable `$page_rel` is overridden for the entire array

Comment (azhavoro) [reply site/layouts/partials/navbar-version-selector.html]: ок

**Review by azhavoro — DISMISSED** [review]

**Comment by azhavoro** [comment]
@TOsmanov Could you please fix linter warnings?

**Review by azhavoro — APPROVED** [review]

**Pull request merged**